opentracing-spring-web-handler-interceptor/src/main/java/io/opentracing/contrib/spring/web/servlet/SpanDecorator.java

```diff
@@ -0,0 +1,107 @@
+     * @param span current span
+     */
+    void onPreHandle(HttpServletRequest httpServletRequest, Object handler, Span span);
+
+    /**
+     * This is called in
+     * {@link org.springframework.web.servlet.HandlerInterceptor#afterCompletion(HttpServletRequest, HttpServletResponse, Object, Exception)}
+     *
+     * @param httpServletRequest request
+     * @param httpServletResponse response
+     * @param handler handler
+     * @param ex exception
+     * @param span current span
+     */
+    void onAfterCompletion(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, Object handler,
+                           Exception ex, Span span);
+
+    /**
+     * Adds standard tags to span. {@link Tags#HTTP_URL}, {@link Tags#HTTP_STATUS}, {@link Tags#HTTP_METHOD} and
+     * {@link Tags#COMPONENT}.
+     */
+    SpanDecorator STANDARD_TAGS = new SpanDecorator() {
+        private final Logger log = Logger.getLogger(SpanDecorator.class.getName());
+
+        @Override
+        public void onPreHandle(HttpServletRequest httpServletRequest, Object handler, Span span) {
+            Tags.COMPONENT.set(span, "java-spring-web-interceptor");
+
+            Tags.HTTP_METHOD.set(span, httpServletRequest.getMethod());
+            //without query params
+            Tags.HTTP_URL.set(span, httpServletRequest.getRequestURL().toString());
+        }
+
+        @Override
+        public void onAfterCompletion(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse,
+                                      Object handler, Exception ex, Span span) {
+            Tags.HTTP_STATUS.set(span, httpServletResponse.getStatus());
+
+            if (ex != null) {
+                Tags.ERROR.set(span, Boolean.TRUE);
+                span.log(logsForException(ex));
+            } else if (httpServletRequest.getAttribute(WebUtils.ERROR_EXCEPTION_ATTRIBUTE) != null) {
+                Tags.ERROR.set(span, Boolean.TRUE);
+                span.log(logsForException((Throwable) httpServletRequest.getAttribute(WebUtils.ERROR_EXCEPTION_ATTRIBUTE)));
+            }
+
+            if (httpServletRequest.getAttribute(WebUtils.ERROR_REQUEST_URI_ATTRIBUTE) != null) {
+                try {
+                    URL url = new URL(httpServletRequest.getRequestURL().toString());
+                    url.getProtocol();
+                    String errorUrl = url.getProtocol() + "://" + url.getHost() + ":" + url.getPort() +
+                            httpServletRequest.getAttribute(WebUtils.ERROR_REQUEST_URI_ATTRIBUTE);
+                    Tags.HTTP_URL.set(span, errorUrl);
+                } catch (MalformedURLException e) {
+                    log.severe("failed to construct error URL");
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        private Map<String, String> logsForException(Throwable throwable) {
+            Map<String, String> errorLog = new HashMap<>();
```

Comment:
You know for sure you will have only 3 entries on the map. Use the constructor to tell it to allocate only 3 positions, instead of the 16 that it does by default.